docs/devenv.rst

```diff
@@ -0,0 +1,593 @@
+    stream = mopidy.stream:Extension
+
+.. warning::
+
+   It's not uncommon to clean up in the Git repo now and then, e.g. by running
+   ``git clean``.
+
+   If you do this, then the :file:`Mopidy.egg-info` directory will be removed,
+   and :mod:`pkg_resources` will no longer know how to locate the "console
+   script" entry point or the bundled Mopidy extensions.
+
+   The fix is simply to run the install command again::
+
+       pip install --editable .
+
+Finally, we can go back to the workspace, again using a virtualenvwrapper
+tool::
+
+   cdproject
+
+
+.. _running-from-git:
+
+Running Mopidy from Git
+=======================
+
+As long as the virtualenv is activated, you can start Mopidy from any
+directory. Simply run::
+
+    mopidy
+
+To stop it again, press :kbd:`Ctrl+C`.
+
+Every time you change code in Mopidy or an extension and want to see it
+live, you must restart Mopidy.
+
+If you wan't to iterate quickly while developing, it may sound a bit tedious to
+restart Mopidy for every minor change. Then it's useful to have tests to
+exercise your code...
+
+
+.. _running-tests:
+
+Running tests
+=============
+
+Mopidy has quite good test coverage, and we would like all new code going into
+Mopidy to come with tests.
+
+.. contents::
+   :local:
+
+
+Test it all
+-----------
+
+You need to know at least one command; the one that runs all the tests::
+
+    tox
+
+This will run exactly the same tests as `Travis CI
+<https://travis-ci.org/mopidy/mopidy>`_ runs for all our branches and pull
+requests. If this command turns green, you can be quite confident that your
+pull request will get the green flag from Travis as well, which is a
+requirement for it to be merged.
+
+As this is the ultimate test command, it's also the one taking the most time to
+run; up to a minute, depending on your system. But, if you have patience, this
+is all you need to know. Always run this command before pushing your changes to
+GitHub.
+
+If you take a look at the tox config file, :file:`tox.ini`, you'll see that tox
+runs tests in multiple environments, including a ``flake8`` environment that
+lints the source code for issues and a ``docs`` environment that tests that the
+documentation can be built. You can also limit tox to just test specific
+environments using the ``-e`` option, e.g. to run just unit tests::
+
+    tox -e py27
+
+To learn more, see the `tox documentation <http://tox.readthedocs.org/>`_ .
+
+
+Running unit tests
+------------------
+
+Under the hood, ``tox -e py27`` will use `pytest <http://pytest.org/>`_ as the
+test runner. We can also use it directly to run all tests::
+
+    py.test
+
+py.test has lots of possibilities, so you'll have to dive into their docs and
+plugins to get full benefit from it. To get you interested, here are some
+examples.
+
+We can limit to just tests in a single directory to save time::
+
+    py.test tests/http/
+
+With the help of the pytest-xdist plugin, we can run tests with four processes
```

Comment:
The --help says:

-n numprocesses shortcut for '--dist=load --tx=NUM*popen'
While http://pytest.org/latest/xdist.html#speed-up-test-runs-by-sending-tests-to-multiple-cpus talks about this feature as way to split tests over multiple CPUs. I guess both are correct in some way: it runs multiple Python processes so that multiple CPUs can contribute to getting the work done.